org.eclipse.xtext/src/org/eclipse/xtext/linking/lazy/LazyLinkingResource.java

```diff
@@ -124,7 +124,7 @@ protected void doLinking() {
 	/**
 	 * @since 2.4
 	 */
-	protected LinkedHashSet<Triple<EObject, EReference, INode>> resolving = Sets.newLinkedHashSet();
+	protected LinkedHashSet<Triple<EObject, EReference, INode>> resolving;
```

Comment:
This is a bit annoying since it never was nullable and now would become nullable.

Reply:
Can you add a method instead?
```suggestion
	protected LinkedHashSet<Triple<EObject, EReference, INode>> resolving = createResolvingSet(true);
	
	protected LinkedHashSet<Triple<EObject, EReference, INode>> createResolvingSet(boolean initial) {
	  ..
	}
```

Your code could override that method and return null if initial is true.
Access to the resolving set can subsequently be null-guarded and `createResolvingSet(false)` can be use to make the instance. 

Thoughts?

Reply:
That works for me, I have changed the PR. What do you think of it now?

Reply:
Looks good.